It looks like this, again, hit:

```
01:56     "stderr_lines": [
01:56         "Traceback (most recent call last):", 
01:56         "  File \"/root/ansible/test/results/.tmp/integration/ansible-runner-04QhQq-ÅÑŚÌβŁÈ/test/integration/targets/ansible-runner/files/adhoc_example1.py\", line 17, in <module>", 
01:56         "    module_args='whoami'", 
01:56         "  File \"/usr/lib/python2.7/site-packages/ansible_runner/interface.py\", line 178, in run", 
01:56         "    r.run()", 
01:56         "  File \"/usr/lib/python2.7/site-packages/ansible_runner/runner.py\", line 120, in run", 
01:56         "    'env': self.config.env}, f, ensure_ascii=False", 
01:56         "  File \"/usr/lib64/python2.7/json/__init__.py\", line 190, in dump", 
01:56         "    fp.write(chunk)", 
01:56         "  File \"/usr/lib64/python2.7/codecs.py\", line 691, in write", 
01:56         "    return self.writer.write(data)", 
01:56         "  File \"/usr/lib64/python2.7/codecs.py\", line 351, in write", 
01:56         "    data, consumed = self.encode(object, self.errors)", 
01:56         "UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 67: ordinal not in range(128)"
01:56     ], 
```

This is known from somewhere, here it is https://github.com/ansible/ansible-runner/issues/443

One minor thing I can do right now is debug the python interpreter in that playbook, then if it's only failing in python 2, then keep hacking to figure out why it's not skipping it, and figure out how we can _really_ make it skip.

--- a/changelogs/fragments/72197-upgrade-test-ansible-runner.yml
+++ b/changelogs/fragments/72197-upgrade-test-ansible-runner.yml
@@ -1,0 +1,4 @@
+minor_changes:
+- ansible-test - Upgrade ansible-runner version used in compatibility tests,
+  remove some tasks that were only needed with older versions, and
+  skip in python2 because ansible-runner is soon dropping it.

--- a/test/integration/targets/ansible-runner/aliases
+++ b/test/integration/targets/ansible-runner/aliases
@@ -1,5 +1,5 @@
 shippable/posix/group3
-skip/python3
+skip/python2  # ansible-runner is for controller and deprecated python2 support
 skip/aix
 skip/osx
 skip/macos

--- a/test/integration/targets/ansible-runner/tasks/adhoc_example1.yml
+++ b/test/integration/targets/ansible-runner/tasks/adhoc_example1.yml
@@ -1,7 +1,5 @@
 - name: execute the script
   command: "'{{ ansible_python_interpreter }}' '{{ role_path }}/files/adhoc_example1.py' '{{ lookup('env', 'OUTPUT_DIR') }}'"
-  environment:
-    AWX_LIB_DIRECTORY: "{{ callback_path }}"
   register: script
 
 - name: parse script output

--- a/test/integration/targets/ansible-runner/tasks/playbook_example1.yml
+++ b/test/integration/targets/ansible-runner/tasks/playbook_example1.yml
@@ -1,7 +1,5 @@
 - name: execute the script
   command: "'{{ ansible_python_interpreter }}' '{{ role_path }}/files/playbook_example1.py' '{{ lookup('env', 'OUTPUT_DIR') }}'"
-  environment:
-    AWX_LIB_DIRECTORY: "{{ callback_path }}"
   register: script
 
 - name: parse script output

--- a/test/integration/targets/ansible-runner/tasks/setup.yml
+++ b/test/integration/targets/ansible-runner/tasks/setup.yml
@@ -1,19 +1,6 @@
-- name: Install docutils
-  pip:
-    name: docutils
-
 - name: Install ansible-runner
   pip:
     name: ansible-runner
-    version: 1.2.0
+    version: 1.4.6
     extra_args:
       -c {{ role_path }}/files/constraints.txt
-
-- name: Find location of ansible-runner installation
-  command: "'{{ ansible_python_interpreter }}' -c 'import os, ansible_runner; print(os.path.dirname(ansible_runner.__file__))'"
-  register: ansible_runner_path
-
-# work around for https://github.com/ansible/ansible-runner/issues/132
-- name: Set callback path to work around ansible-runner bug
-  set_fact:
-    callback_path: ":{{ ansible_runner_path.stdout }}/callbacks"

